### ISSUE TITLE
After upgrade from Flutter 2 to Flutter 3 on running automation, my app immediately crash after launch

### ISSUE THREAD

**Issue by dorbluevine** [flutter/flutter#107833]
We try to run Automation with Webdriver.io, Appium and appium flutter driver. After upgrade from Flutter 2 to Flutter 3 app is crash immediately after launch. It's happen on both Android and iOS (physical and simulator devices). App release with Flutter 2 is work correctly.

Build command:
`flutter build apk --debug --target=./test_driver/build.dart`

Run command:
` npx wdio config/wdio.android.flutter.conf.js --spec tests/flutter_specs/app.test.spec.js`

In Output get an error:
```
DEBUG webdriver: request failed due to response error: unknown error
ERROR webdriver: Request failed with status 500 due to unknown error: An unknown server-side error occurred while processing the command. Original error: Cannot read property 'match' of undefined
ERROR webdriver: unknown error: An unknown server-side error occurred while processing the command. Original error: Cannot read property 'match' of undefined
```

I cleaned npm node modules and package.lock and it didn't help.

My dependencies from package.json:

"dependencies":

    "@wdio/appium-service": "^6.1.16",
    "@wdio/cli": "^6.1.24",
    "@wdio/jasmine-framework": "^6.1.23",
    "@wdio/junit-reporter": "^7.5.3",
    "@wdio/local-runner": "^6.1.24",
    "@wdio/sauce-service": "^6.1.16",
    "@wdio/spec-reporter": "^6.1.23",
    "@wdio/sync": "^6.1.14",
    "appium": "^1.21.0",
    "appium-flutter-driver": "0.0.4",
    "appium-flutter-finder": "^0.1.0",
    "babel-eslint": "^10.1.0",
    "node-fetch": "^2.6.0",
    "webdriverio": "^6.1.24"
    },
    "devDependencies": {
    "npm-bundle": "^3.0.3"
    },
    "bundledDependencies": [
    "@wdio/appium-service",
    "@wdio/cli",
    "@wdio/jasmine-framework",
    "@wdio/junit-reporter",
    "@wdio/local-runner",
    "@wdio/sauce-service",
    "@wdio/spec-reporter",
    "@wdio/sync",
    "babel-eslint",
    "node-fetch",
    "webdriverio"
    ]

**### Update!!**
We clone small flutter project with minimum dependencies: https://github.com/bitrise-dev/flutter-sample-app-hello-world
Before update to flutter 3 all works fine, but after it we get the same error mention before

Any help to solve the issue...

**Comment by exaby73** [comment]
Hello @dorbluevine. Can you provide me with the output of running `flutter doctor -v` and `flutter run -v`?

**Comment by dorbluevine** [comment]
In flutter run -v has no have a problem, 
 flutter doctor -v output is:

```
ilyagarashchenko&development:~/bluevine/sources/flutter-bluevine (master)$ flutter doctor -v

[✓] Flutter (Channel unknown, 3.0.0, on macOS 12.4 21F79 darwin-x64, locale en-IL)
    • Flutter version 3.0.0 at /Users/ilyagarashchenko/bluevine/flutter
    • Upstream repository unknown
    • Framework revision ee4e09cce0 (2 months ago), 2022-05-09 16:45:18 -0700
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2

[!] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/ilyagarashchenko/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.69.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.44.0

[✓] Connected device (3 available)
    • SM N960F (mobile) • 2794a52155217ece • android-arm64  • Android 10 (API 29)
    • macOS (desktop)   • macos            • darwin-x64     • macOS 12.4 21F79 darwin-x64
    • Chrome (web)      • chrome           • web-javascript • Google Chrome 103.0.5060.114

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

**Comment by exaby73** [comment]
@dorbluevine Could you provide steps to reproduce this issue. Unfortunately, I am not very familiar with bitrise

**Comment by exaby73** [comment]
I am able to build and run the minimum dependencies repo you provided... Am I missing any repro steps?

**Comment by dorbluevine** [comment]
**UPDATE:**
It seems we solved the issue, our Appium version was 1.21.0 and appium flutter driver version was 0.0.4, after we upgraded it to Appium 2.0 beta 40 and appium flutter driver to 1.1.0 (and above) - the build was run successfully with Flutter 3. for now we solve it in the minimum dependencies but it a great start. now we just need to configure it in our project and run it.

Anyway, thank you [exaby73](https://github.com/exaby73) for your reply.
